youtube_dl/extractor/vidlii.py

```diff
@@ -0,0 +1,152 @@
+            get_element_by_id('des_text', webpage).strip())
+
+        uploader = str_or_none(
+            self._html_search_regex(
+                r'<div[^>]+class="wt_person"[^>]*>(?:[^<]+)<a href="\/user\/[^>]*?>([^<]*?)<',
+                webpage,
+                'uploader', default=None)) or str_or_none(
+            self._html_search_regex(
+                r'<img src="[^>]+?class=["\']avt2\s*["\'][^>]+?alt=["\']([^"\']+?)["\']',
+                webpage, 'uploader', default=None))
+
+        url = self._html_search_regex(
+            r'videoInfo[\s]*=[\s]*{[^}]*src:[\s]*(?:"|\')([^"]*?)(?:"|\')',
+            webpage, 'url', default=None)
+
+        # get additional properties
+        uploader_url = "https://www.vidlii.com/user/%s" % uploader
+
+        # returns date as YYYYMMDD
+        upload_date = str_or_none(
+            self._html_search_meta('datePublished', webpage, 'upload_date',
+                                   default=False).replace("-",
+                                                          ""))
+
+        categories = self._html_search_regex(
+            r'<div>Category:\s*<\/div>[\s\r]*<div>[\s\r]*<a href="\/videos\?c=[^>]*>([^<]*?)<\/a>',
+            webpage,
+            'categories', default=None)
+        tags = re.findall(r'<a href="/results\?q=[^>]*>[\s]*([^<]*)</a>',
+                          webpage) or None
+        duration = int_or_none(
+            self._html_search_meta('video:duration', webpage, 'duration',
+                                   default=False)) or int_or_none(
+            self._html_search_regex(
+                r'videoInfo[^=]*=[^{]*{[^}]*dur:([^,}]*?),', webpage,
+                'duration', default=None))
+
+        view_count_fb = re.findall(r'<strong>([^<]*?)</strong>',
+                                   get_element_by_class("w_views",
+                                                        webpage))
+        view_count_fb = view_count_fb[0] if view_count_fb else None
+        view_count = int_or_none(self._html_search_regex(
+            r'Views:[^<]*<strong>([^<]*?)<\/strong>', webpage,
+            'view_count', default=None)) or int_or_none(view_count_fb)
+
+        comment_count = int_or_none(self._html_search_regex(
+            r'Comments:[^<]*<strong>([^<]*?)<\/strong>', webpage,
+            'comment_count', default=None)) or int_or_none(
+            self._html_search_regex(
+                r'<span[^>]+id="cmt_num"[^>]*>([^<]+?)<\/span>', webpage,
+                'comment_count', default=None))
+
+        average_rating = float_or_none(
+            self._html_search_regex(
+                r'{[\s\r]*\$\("#rateYo"\).rateYo\({[^}]*rating:\s*([^,]*?),[^}.]*}',
+                webpage, 'average_rating', default=None))
+        thumbnail_link = self._html_search_regex(
+            r'videoInfo[\s]*=[\s]*{[^}]*img:[\s]*(?:"|\')([^"]*?)(?:"|\')',
+            webpage, 'thumbnail', default=None)
+        thumbnail = 'https://www.vidlii.com%s' % thumbnail_link
```

Comment:
`urljoin`.

Reply:
FIXED: Use of ```urljoin``` from utils.py. Same changes made for "uploader" attribute.